src/main/java/uk/gov/hmcts/reform/ccd/service/remote/DisposeTasksRemoteOperation.java

```diff
@@ -1,73 +1,70 @@
 
-import com.google.gson.Gson;
-import jakarta.ws.rs.core.Response;
+import feign.Response;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.ccd.data.model.CaseData;
 import uk.gov.hmcts.reform.ccd.data.tm.DeleteCaseTasksAction;
 import uk.gov.hmcts.reform.ccd.data.tm.DeleteTasksRequest;
-import uk.gov.hmcts.reform.ccd.parameter.ParameterResolver;
+import uk.gov.hmcts.reform.ccd.exception.TasksDeletionException;
+import uk.gov.hmcts.reform.ccd.service.remote.clients.TasksClient;
+import uk.gov.hmcts.reform.ccd.util.SecurityUtil;
 import uk.gov.hmcts.reform.ccd.util.log.TasksDeletionRecordHolder;
 
-import static uk.gov.hmcts.reform.ccd.util.RestConstants.DELETE_TASKS_PATH;
-
 @Service
 @Slf4j
 public class DisposeTasksRemoteOperation implements DisposeRemoteOperation {
 
-    private final ParameterResolver parameterResolver;
-
-    private final Gson gson = new Gson();
+    private final SecurityUtil securityUtil;
 
     private TasksDeletionRecordHolder tasksDeletionRecordHolder;
 
-    private final CcdRestClientBuilder ccdRestClientBuilder;
+    private final TasksClient tasksClient;
 
     @Autowired
-    public DisposeTasksRemoteOperation(final CcdRestClientBuilder ccdRestClientBuilder,
-                                       final ParameterResolver parameterResolver,
+    public DisposeTasksRemoteOperation(final TasksClient tasksClient,
+                                       final SecurityUtil securityUtil,
                                        final TasksDeletionRecordHolder tasksDeletionRecordHolder) {
-        this.ccdRestClientBuilder = ccdRestClientBuilder;
-        this.parameterResolver = parameterResolver;
+        this.tasksClient = tasksClient;
+        this.securityUtil = securityUtil;
         this.tasksDeletionRecordHolder = tasksDeletionRecordHolder;
     }
 
+    @SuppressWarnings("java:S1135")
     @Override
     public void delete(final CaseData caseData) {
         final String caseRef = caseData.getReference().toString();
         try {
             final DeleteTasksRequest tasksDeletePostRequest =
-                    new DeleteTasksRequest(new DeleteCaseTasksAction(caseRef));
+                new DeleteTasksRequest(new DeleteCaseTasksAction(caseRef));
 
-            final String requestDeleteBody = gson.toJson(tasksDeletePostRequest);
 
-            final Response taskDeleteResponse = deleteTasks(requestDeleteBody);
+            final Response taskDeleteResponse = deleteTasks(tasksDeletePostRequest);
 
-            logTasksDisposal(caseRef, taskDeleteResponse);
+            tasksDeletionRecordHolder.setCaseTasksDeletionResults(caseRef, taskDeleteResponse.status());
 
-            if (taskDeleteResponse.getStatus() != 201) {
-                log.error("Error in deleting tasks for case : Case Ref = {} ", caseRef);
-            }
+            if (taskDeleteResponse.status() != 201) {
```

Comment:
Is deletion response 201? shouldn't that be 204? Maybe more generically better to use something like `is2xxSuccessful()`